### ISSUE TITLE
Tracking Issue: drop support for OpenSSL

### ISSUE THREAD

**Issue by marten-seemann** [libp2p/go-libp2p#1954]
Motivation: see #1952.

TLDR: OpenSSL support doesn't seem to provide any (significant) performance benefits outside of microbenchmarks. To stop it from continuing to consume a disproportionately of maintainer attention we should remove it.

Done criteria:
- [x] remove OpenSSL code paths from go-libp2p: https://github.com/libp2p/go-libp2p/pull/1953
- [x] archive https://github.com/libp2p/go-openssl
- [x] write this up in the release notes for the next release
- [ ] maybe write a blog post? It might be interesting to explain what benchmarks we did and how we arrived at our conclusion. How to migrate the IPFS bootstrappers (https://github.com/ipfs/kubo/issues/9507) away from RSA might also be an interesting story.

**Comment by marten-seemann** [comment]
Closing this as completed. The PR has been merged, the repo archived and I wrote up release notes. Not sure if we want to actually write a blog post, given that we don't expect any measurable performance impact.
